sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -160,19 +160,22 @@ private[sql] class DefaultSource
           val job = Job.getInstance(conf)
           FileInputFormat.setInputPaths(job, file.filePath)
 
-          val inputFormat = new OrcNewInputFormat
           val fileSplit = new FileSplit(
             new Path(new URI(file.filePath)), file.start, file.length, Array.empty
           )
-
-          val attemptId = new TaskAttemptID(new TaskID(new JobID(), TaskType.MAP, 0), 0)
-          val hadoopAttemptContext = new TaskAttemptContextImpl(conf, attemptId)
-          inputFormat.createRecordReader(fileSplit, hadoopAttemptContext)
+          // Create RecordReader here. This will help in getting
+          // ObjectInspector during recordReader creation itself and can
+          // avoid NN call in unwrapOrcStructs to get ObjectInspector for the
+          // file. Would be helpful for partitioned datasets.
```

Comment:
It would be good to emphasize that this `OrcRecordReader` is a customized one. People who are not familiar with ORC may think it's the default ORC record reader.